src/worker.c

```diff
@@ -113,47 +132,45 @@ static CURLMcode init(CURLM *cm, char *method, char *url, int timeout_millisecon
 	(void)pushJsonbValue(&response_headers, WJB_BEGIN_OBJECT, NULL);
 	cdata->response_headers = response_headers;
 	cdata->id = id;
-	cdata->request_headers = request_headers;
-
-	request_headers = curl_slist_append(request_headers, "User-Agent: pg_net/" EXTVERSION);
+	cdata->request_headers = curl_slist_append(request_headers, "User-Agent: pg_net/" EXTVERSION);
```

Comment:
To avoid overwriting an existing non-empty list on failure, the new list should be returned to a temporary variable which can be tested for NULL before updating the original list pointer